gobblin-compaction/src/main/java/gobblin/compaction/Dataset.java

```diff
@@ -237,10 +237,22 @@ public void setJobProps(State jobProps) {
     return this.throwables;
   }
 
+  /**
+   * Record a {@link Throwable} in a {@link Dataset}.
+   */
   public synchronized void addThrowable(Throwable t) {
     this.throwables.add(t);
   }
 
+  /**
+   * Skip the {@link Dataset} by setting its {@link DatasetState} to {@link DatasetState#COMPACTION_COMPLETE},
+   * and record the given {@link Throwable} in the {@link Dataset}.
+   */
+  public void skip(Throwable t) {
+    this.state = DatasetState.COMPACTION_COMPLETE;
```

Comment:
Does `this.state` also needs to be guarded? If so, you probably should use a `AtomicReference` instead.